prow/cmd/hook/main.go

```diff
@@ -220,5 +220,20 @@ func main() {
 	// Serve plugin help information from /plugin-help.
 	http.Handle("/plugin-help", pluginhelp.NewHelpAgent(pluginAgent, githubClient))
 
-	logrus.Fatal(http.ListenAndServe(":"+strconv.Itoa(o.port), nil))
+	httpServer := &http.Server{Addr: ":" + strconv.Itoa(o.port), Handler: nil}
+
+	// Shutdown gracefully on SIGTERM or SIGINT
+	go func() {
+		sig := make(chan os.Signal, 1)
+		signal.Notify(sig, os.Interrupt, syscall.SIGTERM)
+		select {
+		case <-sig:
+			logrus.Info("Hook is shutting down...")
+			ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
nit: just s/timeout/180 * time.Second here, as it's not referenced anywhere else

Reply:
this should almost definitely be configurable with a default (make it a flag probably)